pkg/csi-common/utils.go

```diff
@@ -21,12 +21,17 @@ import (
 	"strings"
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
 )
 
 func ParseEndpoint(ep string) (string, string, error) {
 	if strings.HasPrefix(strings.ToLower(ep), "unix://") || strings.HasPrefix(strings.ToLower(ep), "tcp://") {
 		s := strings.SplitN(ep, "://", 2)
-		return s[0], s[1], nil
+		if s[1] != "" {
```

Comment:
validate len(s) too

Reply:
There is a lot of validation code missing. CSI specs defined length limits & restrictions on various parameters. Will add it, as part of it.